wa/workloads/lmbench/__init__.py

```diff
@@ -0,0 +1,170 @@
+                  Specifies an lmbench test to run.
+                  '''),
+        Parameter('stride', kind=list_or_integer, default=[128],
+                  description='''
+                  Stride for lat_mem_rd test. Workload will iterate over one or
+                  more integer values.
+                  '''),
+        Parameter('thrash', kind=bool, default=True,
+                  description='Sets -t flag for lat_mem_rd_test'),
+        Parameter('size', kind=list_or_string, default="4m",
+                  description='Data set size for lat_mem_rd bw_mem tests.'),
+        Parameter('mem_category', kind=list_or_string,
+                  default=('rd', 'wr', 'cp', 'frd', 'fwr', 'fcp', 'bzero', 'bcopy'),
+                  description='List of memory catetories for bw_mem test.'),
+        Parameter('parallelism', kind=int, default=None,
+                  description='Parallelism flag for tests that accept it.'),
+        Parameter('warmup', kind=int, default=None,
+                  description='Warmup flag for tests that accept it.'),
+        Parameter('repetitions', kind=int, default=None,
+                  description='Repetitions flag for tests that accept it.'),
+        Parameter('force_abi', kind=str, default=None,
+                  description='''
+                  Override device abi with this value. Can be used to force
+                  arm32 on 64-bit devices.
+                  '''),
+        Parameter('run_timeout', kind=int, default=900,
+                  description="""
+                  Timeout for execution of the test.
+                  """),
+        Parameter('loops', kind=int, default=1, constraint=lambda x: x > 0,
+                  description="""
+                  Specifies the number of times the benchmark will be run in a
+                  "tight loop", i.e. without performaing setup/teardown
+                  inbetween. This parameter is distinct from "repetitions", as
+                  the latter takes place within the benchmark and produces a
+                  single result.
+                  """),
+        Parameter('cpus', kind=cpu_mask, default=0, aliases=['taskset_mask'],
+                  description="""
+                  Specifies the CPU mask the benchmark process will be pinned to.
+                  """),
+    ]
+
+    def setup(self, context):
+
+        abi = self.target.abi
+        if self.force_abi:
+            abi = self.force_abi
+
+        # self.test has been pre-validated, so this _should_ only fail if there's an abi mismatch
+        host_exe = context.resolver.get(Executable(self, abi, self.test))
+        self.target_exe = self.target.install(host_exe)
+        self.commands = []
+
+        setup_test = getattr(self, '_setup_{}'.format(self.test))
+        setup_test()
+
+        for _ in range(self.times):
+            for command in self.commands:
+                self.target.execute(command, timeout=self.run_timeout)
+
+    def update_result(self, context):
```

Comment:
This method is not used in WA3, valid methods are  `extract_results` and `update_output`. Although it looks like the functionality here is duplicated in your `extract_results` method so this can probably just be removed.